src/strings.js

```diff
@@ -1,8 +1,11 @@
+const util = require("./utilities");
+
 var engine = {};
 
 function ensureStringSingleton(x){
-  if(x.length == 1 && typeof x[0] === "string") {
-    return x[0];
+  let d;
+  if(x.length == 1 && typeof (d=util.valData(x[0])) === "string") {
```

Comment:
Any reason why not checking/allowing typeof valData(x) == 'string'?

Reply:
The expected x is an array.  It should be an array with one string (or ResourceNode).

Reply:
but the math.ensureSingletonNumber(?) checks the input itself as well, I would think they need to be consistent either way.

Reply:
ensureNumberSingleton and ensureStringSingleton were written by two different people with possibly different use cases in mind.  It turns out that ensureNumberSingleton sometimes gets a plain number not wrapped in an array.  However, I noticed a mistake in my changes to ensureNumberSingleton, which I have now fixed.